test/runtests.js

```diff
@@ -391,3 +391,73 @@ test('JSON with serializer',
 
   handler(e, testReq(), res, testNext);
 });
+
+test('JSON with serializer with access to error object', 
+    function (t) {
+
+  var shutdown = function shutdown() {},
+    e = (function () {
+      var err = new Error();
+      err.status = 400;
+      ['code', 'name', 'type', 'details'].forEach(function(prop) { err[prop] = 'foo'; });
+      return err;
+    }()),
+    handler = createHandler({
+      shutdown: shutdown,
+      serializer: function(err) { return err }
+    }),
+    res = testRes({
+      send: function send(obj) {
+        var propertiesPass = ['code', 'name', 'type', 'details'].every(function(prop) {
+          return obj[prop] === 'foo';
+        });
+        t.ok(propertiesPass,
+          'Should be able to write custom serializer with access to properties of client errors.');
+        t.end();
+      },
+      format: function format (types) {
+        return types['json']();
+      }
+    });
+
+    handler(e, testReq(), res, testNext);
+  });
+
+test('JSON with intelligent serializer that utilizes isClientError internally', 
```

Comment:
`isClientError()` already has tests. This test seems to be testing... what? itself...?